test/src/unit-udt_macro.cpp

```diff
@@ -59,6 +59,42 @@ class person_with_private_data
     NLOHMANN_DEFINE_TYPE_INTRUSIVE(person_with_private_data, age, name, metadata)
 };
 
+class person_with_private_data_2
```

Comment:
Do you actually test how JSON objects with missing values are serialized?

Reply:
Test added.